core/variables.js

```diff
@@ -287,23 +287,37 @@ Blockly.Variables.renameVariable = function(workspace, variable,
  * @param {string} defaultText The default value to show in the prompt's field.
  * @param {function(?string)} callback A callback. It will be passed the new
  *     variable name, or null if the user picked something illegal.
+ * @param {string} opt_type Optional type of variable, like 'string' or 'list'.
  */
-Blockly.Variables.promptName = function(promptText, defaultText, callback) {
+Blockly.Variables.promptName = function(promptText, defaultText, callback, opt_type) {
   Blockly.prompt(promptText, defaultText, function(newVar) {
     // Merge runs of whitespace.  Strip leading and trailing whitespace.
     // Beyond this, all names are legal.
     if (newVar) {
-      newVar = newVar.replace(/[\s\xa0]+/g, ' ').replace(/^ | $/g, '');
-      if (newVar == Blockly.Msg.RENAME_VARIABLE ||
-          newVar == Blockly.Msg.NEW_VARIABLE) {
-        // Ok, not ALL names are legal...
-        newVar = null;
-      }
+      newVar = Blockly.Variables.validateName_(newVar, opt_type);
     }
     callback(newVar);
   });
 };
 
+/**
+ * Validate the variable name provided by the user.
+ * @param {string} name The user-provided name of the variable.
+ * @param {string} opt_type Optional type of variable, like 'string' or 'list'.
+ * @return {string} The validated and possibly transformed name of the variable.
+ */
+Blockly.Variables.validateName_ = function(name, opt_type) {
+  if (!opt_type || !opt_type == Blockly.BROADCAST_MESSAGE_VARIABLE_TYPE) {
```

Comment:
hmmmm can you use `x != y` instead of `!x == y`. I'm surprised by stickiness every time.

Reply:
done

Reply:
Not stickiness, the code was doing what you you would think and it just happened to work out:
<img width="194" alt="screen shot 2017-12-21 at 3 34 59 pm" src="https://user-images.githubusercontent.com/2879020/34277791-88f1bc1c-e664-11e7-8cf4-3a1da25a4c7b.png">